bot/code_review_bot/report/phabricator.py

```diff
@@ -70,25 +78,39 @@ def publish(self, issues, revision, task_failures):
             # Always publish comment summarizing issues
             self.publish_comment(revision, issues_only, patches, task_failures)
 
-            # Also publish build issues
-            if self.publish_build_errors:
-                self.publish_harbormaster_build_errors(revision, build_errors)
+            # Publish all errors outside of the patch as lint issues
+            lint_issues = [
+                issue
+                for issue in issues_only
+                if not revision.contains(issue) and issue.level == Level.Error
+            ]
+
+            # Also publish build errors as Phabricator unit result
+            unit_issues = build_errors if self.publish_build_errors else []
+
+            # Publish on Harbormaster all at once
+            self.publish_harbormaster(revision, lint_issues, unit_issues)
         else:
             logger.info("No issues to publish on phabricator")
 
         return issues, patches
 
-    def publish_harbormaster_build_errors(self, revision, issues):
+    def publish_harbormaster(
+        self, revision, lint_issues: Issues = [], unit_issues: Issues = []
+    ):
         """
-        Publish build errors through Phabricator UnitResult
+        Publish issues through HarborMaster
+        either as lint results or unit tests results
         """
-        if not issues:
-            logger.info("No build errors encountered")
+        if not lint_issues and not unit_issues:
+            logger.info("No issues to publish as Phabricator lint or unit results")
             return
+
         self.api.update_build_target(
             revision.build_target_phid,
-            BuildState.Fail,
-            unit=[issue.as_phabricator_unitresult() for issue in issues],
+            state=BuildState.Work,
```

Comment:
Why Work and not Fail?

Reply:
Fail is sent afterward by the workflow. It should not be the responsibility of the phabricator reporter to give the final state (as it's set in stone once fail or pass).

Currently, a revision with a build error gets 2 Fail status, which is weird

Reply:
> Fail is sent afterward by the workflow

Do you mean https://github.com/mozilla/code-review/blob/d570e38cea78f7479d5d0f6b81934a6689ef169a/bot/code_review_bot/workflow.py#L200? Does this one cover the build failure case?

Reply:
Yes, `nb_publishable` will be positive for any in patch warnings or errors anywhere